util-core/src/main/scala/com/twitter/util/Future.scala

```diff
@@ -339,6 +339,27 @@ def join[%s](%s): Future[(%s)] = join(Seq(%s)) map { _ => (%s) }""".format(
   def join[A](fs: JList[Future[A]]): Future[Unit] = Futures.join(fs)
 
   /**
+   * Take a sequence and sequentially apply a function f to each item.
+   * Then return all future results as a single Future[Seq[_]].
+   * 
+   * If during execution any f() throws an exception that exception will be returned and 
+   * the remaining futures will not be processed.
+   * 
+   * usage:
+   *   Future.traverseSequentially(Seq(1,2,3))(deleteItem)
+   * 
+   * @param as a sequence of A that will have f applied to each item sequentially
+   * @return a `Future[Seq[B]]` containing the results of f being applied to every item in as
+   */
+  def traverseSequentially[A,B](as: Seq[A])(f: A => Future[B]): Future[Seq[B]] =
+    as.foldLeft(Future(Vector.empty[B])) { (resultsFuture, nextItem) =>
```

Comment:
I'm not convinced that the shaved allocation is worth any extra lines of code so I'm still happy with the inline Vector.empty, but your call. If you go the shaving route, we should be able to drop the cast by adding the type param to the foldleft though.